docs/guides/how_to_guides/creating_and_editing_expectations/how_to_create_an_expectation_suite_with_the_user_configurable_profiler.rst

```diff
@@ -0,0 +1,250 @@
+
+                import great_expectations as ge
+                from great_expectations.profile.user_configurable_profiler import UserConfigurableProfiler
+                context = ge.data_context.DataContext(
+                    context_root_dir='path/to/my/context/root/directory/great_expectations'
+                )
+
+            Create an in-code Data Context using these instructions: :ref:`How to instantiate a Data Context without a yml file <how_to_guides__configuring_data_contexts__how_to_instantiate_a_data_context_without_a_yml_file>`
+
+
+        2. Create a new Expectation Suite
+
+            .. code-block:: python
+
+                expectation_suite_name = "insert_your_expectation_suite_name_here"
+                suite = context.create_expectation_suite(
+                    expectation_suite_name
+                )
+
+
+        3. Construct batch_kwargs and get a Batch
+
+            ``batch_kwargs`` describe the data you plan to validate. In the example below we use a Datasource configured for SQL Alchemy. This will be a little different for SQL, Pandas, and Spark  back-ends - you can get more info on the different ways of instantiating a batch here: :ref:`How to create a batch <how_to_guides__creating_batches>`
+
+            .. code-block:: python
+
+                batch_kwargs = {
+                    "datasource": "insert_your_datasource_name_here",
+                    "schema": "your_schema_name",
+                    "table": "your_table_name",
+                    "data_asset_name": "your_table_name",
+                }
+
+
+            Then we get the Batch via:
+
+            .. code-block:: python
+
+                batch = context.get_batch(
+                    batch_kwargs=batch_kwargs,
+                    expectation_suite_name=suite
+                )
+
+
+        4. Check your data
+
+            You can check that the first few lines of your Batch are what you expect by running:
+
+            .. code-block:: python
+
+                batch.head()
+
+
+        5. Instantiate a UserConfigurableProfiler
+
+            Next, we instantiate a UserConfigurableProfiler, passing in our batch
+
+            .. code-block:: python
+
+                profiler = UserConfigurableProfiler(dataset=batch)
+
+
+        6. Use the profiler to build a suite
+
+            Now that we have our profiler set up with our batch, we can use the build_suite method on the profiler. This will print a list of all the expectations created by column, and return the Expectation Suite object.
+
+            .. code-block:: python
+
+                suite = profiler.build_suite()
+
+        7. (Optional) Running validation, saving the suite and building Data Docs
+
+            If you'd like, you can validate the your data with the new suite, save your Expectation Suite and build Data Docs to take a closer look
+
+            .. code-block:: python
+                # We need to re-create our batch to link the batch with our new suite
```

Comment:
```suggestion
                
                # We need to re-create our batch to link the batch with our new suite
```

Lol, rst woes again... I was reviewing using the rendered page, which didn't have the code block, and was going to say "Would be nice to have code snippet", but then realized you did. :P Just needed an empty line after the `python` directive.

Reply:
Ah - great catch, thanks!